picky/src/x509/pkcs7/authenticode.rs

```diff
@@ -0,0 +1,1886 @@
+                                                Cert::from_der(&certificate.0).ok()
+                                            }
+                                            CertificateChoices::Other(_) => None,
+                                        })
+                                        .collect::<Vec<Cert>>();
+
+                                    if check_if_kp_time_stamping_present(&certificates)
+                                        && !kp_lifetime_signing_is_present
+                                    {
+                                        // check if certificates in the timestamp have the right chain
+                                        if let Some(leaf) = certificates.get(0) {
+                                            let timestamp_chain_validator = leaf.verifier();
+                                            let timestamp_chain = certificates
+                                                .iter()
+                                                .filter(|cert| cert.subject_name() != leaf.subject_name())
+                                                .cloned()
+                                                .collect::<Vec<Cert>>();
+
+                                            let timestamp_chain_validator = timestamp_chain_validator
+                                                .require_chain_check()
+                                                .chain(timestamp_chain.iter());
+
+                                            let timestamp_chain_validator = timestamp_chain_validator
+                                                .ignore_not_after_check()
+                                                .ignore_not_before_check()
+                                                .chain_should_contains_root_certificate(false);
+
+                                            timestamp_chain_validator
+                                                .verify()
+                                                .map_err(AuthenticodeError::InvalidTimestampCert)?;
+
+                                            #[cfg(feature = "ctl")]
+                                            {
+                                                let ca_name = get_ca_name(&certificates).unwrap();
+                                                self.verify_ca_certificate_against_ctl(&ca_name)?;
+                                            }
+                                        }
+
+                                        return Ok(());
+                                    }
+                                }
+                                UnsignedAttributeValue::CounterSign(_) => {}
+                            };
+                        }
+                    }
+                }
+
+                Err(AuthenticodeError::CertError(err))
+            }
+        }
+    }
+
+    #[cfg(feature = "ctl")]
+    fn verify_ca_certificate_against_ctl(&self, ca_name: &DirectoryName) -> AuthenticodeResult<()> {
```

Comment:
Please, add a link to the reference (RFC, specification, source code) you used when implementing this function